core/src/main/groovy/noe/rhel/server/httpd/HttpdBaseOS.groovy

```diff
@@ -0,0 +1,114 @@
+package noe.rhel.server.httpd
+
+import groovy.util.logging.Slf4j
+import noe.common.utils.FileStateVault
+import noe.common.utils.JBFile
+import noe.common.utils.Library
+import noe.common.utils.Version
+import noe.server.Httpd
+/**
+ * RHEL BaseOS Httpd server
+ * @author Jan Onderka <jonderka@redhat.com> *
+ */
+@Slf4j
+class HttpdBaseOS extends Httpd {
+  protected String serviceName
+  private FileStateVault fileStateVault
+
+  HttpdBaseOS(String basedir, version) {
+    super(basedir, version)
+    this.serviceName = "httpd"
+    this.basedir = "/etc/${serviceName}"
+    this.fileStateVault = new FileStateVault()
+
+    setDefault()
+    logDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void setDefault() {
+    super.setDefault()
+    this.configDirs = ['/conf', '/conf.d']
+    if (version >= new Version("2.4")) {
+      this.configDirs.add('/conf.modules.d')
+    }
+    this.modulesDir = this.basedir + "/modules"
+    this.logDirs = ['/logs']
+    // what path will be taken as relative for star/stop cmd
+    this.binPath = '/'
+    if (platform.isRHEL8() || platform.isRHEL7()) {
+      this.start = ['systemctl', 'start', serviceName]
+      this.stop = ['systemctl', 'stop', serviceName]
+    } else if (platform.isRHEL6()) {
+      this.start = ['service', serviceName, 'start']
+      this.stop = ['service', serviceName, 'stop']
+    } else {
+      log.error("Couldn't determine supported RHEL version.")
+      this.start = null
```

Comment:
@jstefl @jan-onderka Can't we make this the default..? The way I see it:

1. We either assume RHEL 8+ and other OSes have systemctl, and make it default, or
1. We throw an exception here and terminate

If we put just null here, that makes it more difficult to find the error later on. Why would we want to continue in this case if we can't start or stop the server?

Reply:
Throw an exception implemented in commit " update 4"